### DISCUSSION TITLE
Implementing nonce with styled-jsx for Content Security Policy

### DISCUSSION THREAD

**Discussion by KarthikeyanRanasthala** [vercel/next.js#15056] [Help]
The implementation mentioned [here](https://github.com/vercel/styled-jsx#content-security-policy) in styled-jsx docs uses flushToReact/flushToHTML which used react-dom/server and styled-jsx/server

As next.js has inbuilt support for SSR and styled-jsx, the above mentioned implementation makes no sense. And this [example](https://github.com/vercel/next.js/tree/canary/examples/with-strict-csp) doesn't cover nonce implementation

FYI, I've tried adding nonce to all individual <style jsx> and its getting stripped somehow during the build process. 

Having a strong CSP is necessary to mitigate XSS and packet sniffing attacks. So let's discuss on this and figure out a way to do this.

**Comment by KarthikeyanRanasthala** [comment]
@timneutkens Any word here?

**Reply by KarthikeyanRanasthala** [reply]
I figured it out. Check this out, https://github.com/KarthikeyanRanasthala/styled-jsx-with-csp/

@timneutkens @leerob @ijjk What should I do here? Add to next.js/examples or add as a feature to next _document itself?
I think this would be much helpful if we add this to next.js itself as a feature. So that everyone can start using it immediately in future next.js versions.

Let me know what suits the most. I can open a PR for it, depending on your views.

**Reply by leerob** [reply]
Feel free to open a PR to add an example. Thank you!

**Reply by Hellstellar** [reply]
@KarthikeyanRanasthala The above example works flawlessly thanks you so much, I wasn't able to find the solution anywhere until I came here, great work.

**Reply by ganesham019** [reply]
@KarthikeyanRanasthala , is still the 'flush' is working fine ?
I am trying to add the 'nonce' to all the style tags, that are generated from styledJsx ?

**Reply by KarthikeyanRanasthala** [reply]
Hey @ganesham019 I haven't had a look at this in a long time, sorry, I can't help at the moment.
